shared-bindings/audiopwmio/AudioOut.c

```diff
@@ -0,0 +1,294 @@
+              (mp_obj_t)&mp_const_none_obj,
+              (mp_obj_t)&mp_const_none_obj},
+};
+
+//|   .. method:: pause()
+//|
+//|     Stops playback temporarily while remembering the position. Use `resume` to resume playback.
+//|
+STATIC mp_obj_t audiopwmio_audioout_obj_pause(mp_obj_t self_in) {
+    audiopwmio_audioout_obj_t *self = MP_OBJ_TO_PTR(self_in);
+    check_for_deinit(self);
+
+    if (!common_hal_audiopwmio_audioout_get_playing(self)) {
+        mp_raise_RuntimeError(translate("Not playing"));
+    }
+    common_hal_audiopwmio_audioout_pause(self);
+    return mp_const_none;
+}
+MP_DEFINE_CONST_FUN_OBJ_1(audiopwmio_audioout_pause_obj, audiopwmio_audioout_obj_pause);
+
+//|   .. method:: resume()
+//|
+//|     Resumes sample playback after :py:func:`pause`.
+//|
+STATIC mp_obj_t audiopwmio_audioout_obj_resume(mp_obj_t self_in) {
+    audiopwmio_audioout_obj_t *self = MP_OBJ_TO_PTR(self_in);
+    check_for_deinit(self);
+
+    if (common_hal_audiopwmio_audioout_get_paused(self)) {
+        common_hal_audiopwmio_audioout_resume(self);
+    }
+
+    return mp_const_none;
+}
+MP_DEFINE_CONST_FUN_OBJ_1(audiopwmio_audioout_resume_obj, audiopwmio_audioout_obj_resume);
+
+//|   .. attribute:: paused
+//|
+//|     True when playback is paused. (read-only)
+//|
+STATIC mp_obj_t audiopwmio_audioout_obj_get_paused(mp_obj_t self_in) {
+    audiopwmio_audioout_obj_t *self = MP_OBJ_TO_PTR(self_in);
+    check_for_deinit(self);
+    return mp_obj_new_bool(common_hal_audiopwmio_audioout_get_paused(self));
+}
+MP_DEFINE_CONST_FUN_OBJ_1(audiopwmio_audioout_get_paused_obj, audiopwmio_audioout_obj_get_paused);
+
+const mp_obj_property_t audiopwmio_audioout_paused_obj = {
+    .base.type = &mp_type_property,
+    .proxy = {(mp_obj_t)&audiopwmio_audioout_get_paused_obj,
+              (mp_obj_t)&mp_const_none_obj,
+              (mp_obj_t)&mp_const_none_obj},
+};
+
+STATIC const mp_rom_map_elem_t audiopwmio_audioout_locals_dict_table[] = {
+    // Methods
+    { MP_ROM_QSTR(MP_QSTR_deinit), MP_ROM_PTR(&audiopwmio_audioout_deinit_obj) },
+    { MP_ROM_QSTR(MP_QSTR___enter__), MP_ROM_PTR(&default___enter___obj) },
+    { MP_ROM_QSTR(MP_QSTR___exit__), MP_ROM_PTR(&audiopwmio_audioout___exit___obj) },
+    { MP_ROM_QSTR(MP_QSTR_play), MP_ROM_PTR(&audiopwmio_audioout_play_obj) },
+    { MP_ROM_QSTR(MP_QSTR_stop), MP_ROM_PTR(&audiopwmio_audioout_stop_obj) },
+    { MP_ROM_QSTR(MP_QSTR_pause), MP_ROM_PTR(&audiopwmio_audioout_pause_obj) },
+    { MP_ROM_QSTR(MP_QSTR_resume), MP_ROM_PTR(&audiopwmio_audioout_resume_obj) },
+
+    // Properties
+    { MP_ROM_QSTR(MP_QSTR_playing), MP_ROM_PTR(&audiopwmio_audioout_playing_obj) },
+    { MP_ROM_QSTR(MP_QSTR_paused), MP_ROM_PTR(&audiopwmio_audioout_paused_obj) },
+};
+STATIC MP_DEFINE_CONST_DICT(audiopwmio_audioout_locals_dict, audiopwmio_audioout_locals_dict_table);
+
+const mp_obj_type_t audiopwmio_audioout_type = {
+    { &mp_type_type },
+    .name = MP_QSTR_AudioOut,
```

Comment:
Could you rename this to PWMAudioOut or PulseAudioOut too? That way it'll print differently than audioio.AudioOut.

Reply:
I realize this is super nitpicky but please rename the files and function names too from `audioout` to `pwmaudioout`. Future us will thank you for the consistency.

Reply:
Does this latest commit get the renames you were looking for, @tannewt ?